go.mod

```
@@ -2,7 +2,7 @@ module github.com/etcd-io/auger
 
 go 1.23
 
-toolchain go1.23.0
+toolchain go1.23.1
```

```suggestion
go 1.23.1
```

Maybe let `toolchain` default to the same as the `go` will be better.

---

I would prefer to keep aligned with how the broader etcd project uses the `go` and `toolchain` directive. 

If we bump the `go` directive we force people to use a specific patch version of go in order to use `auger` which is not desirable.

Ref https://go.dev/doc/toolchain

> Modules that are dependencies of other modules may need to set a minimum Go version requirement lower than the preferred toolchain to use when working in that module directly. In this case, the toolchain line in go.mod or go.work sets a preferred toolchain that takes precedence over the go line when the go command is deciding which toolchain to use.

---

thank you for explaining

/lgtm